source/MissionAction.h

```diff
@@ -72,26 +74,30 @@ class MissionAction {
 	std::string system;
 	LocationFilter systemFilter;
 	
-	std::string logText;
-	std::map<std::string, std::map<std::string, std::string>> specialLogText;
-	
 	std::string dialogText;
 	const Phrase *stockDialogPhrase = nullptr;
 	Phrase dialogPhrase;
 	
 	const Conversation *stockConversation = nullptr;
 	Conversation conversation;
 	
-	std::map<const GameEvent *, std::pair<int, int>> events;
 	std::map<const Outfit *, int> gifts;
 	std::map<const Outfit *, int> requiredOutfits;
+	
+	GameAction actions;
```

Comment:
The MissionAction now has a GameAction inside it. By itself this is fine, but I was expecting that the MissionAction would actually inherit from the GameAction class instead of embedding a GameAction class inside it (Similar to how a Ship inherits from the Body class).

If you have good reasons for choosing this pattern, then that is fine for me, but maybe add a comment on why the "contain pattern" instead of the "inherit pattern" was choosen.

Reply:
The difference between Ship/Body and MissionAction/GameAction is that Body gives new functions to Ship, but MissionAction has functions that go by the same name as those in GameAction while doing the same thing as them and more. I did actually have MissionAction inherit GameAction at first, but this seemed to result in duplicating code because of the functions of the same name. (Perhaps there’s simply some process to avoid that that I don’t know about, though.)

Reply:
Functions that have the same name but do something more sound like something for which the "super()" functionality in Java was invented. Super() allows a function to call the ancestors version of itself and then add something more around it (before or after calling super()). C++ also has something similar, but you need to specify the name of the base class due to multiple-inheritance.

Reply:
Yeah, I recall Super() in Java. So how might that be done here?

Reply:
My first thoughts on this are:
- make the class MissionAction inherit from GameAction.
- make LoadSingle() in GameAction.h virtual and override it in MissionAction.
   - process MissionAction-specifics in MissionAction::LoadSingle()
   - call GameAction::LoadSingle() from MissionAction::LoadSingle() for every keyword not handled in MissionAction.
   - Keep Load() only in GameAction (and not virtual).
- make Save() in GameAction.h virtual and override it in MissionAction.
   - call GameAction::Save() from MissionAction::Save()
- make Do() in GameAction.h virtual and override it in MissionAction.
   - Call GameAction::Do() from MissionAction::Do() after or before Doing the MissionAction specific things.
- Instantiate() cannot be virtual and overriden since they return different objects. There you could look at using a constructor for the returned object that calls baseclass constructors.

Reply:
It still needs cleaned up some, but how does the recent commit (e13190b) look?

Reply:
I briefly reviewed the latest version of the PR (at commit fc8478a86cf7fcb1720fe5193dd3ac2a5386a0e5) and my first impression is that it looks quite good.
I didn't look into all the details yet; that requires me to review a bit more in-depth, but you can consider this conversation done/handled.